pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts

```diff
@@ -103,3 +103,28 @@ export const download = (blob: Block): string => {
 
   return url;
 };
+
+export const getOverlappingBlocks = (blockPools: { [source: string]: BlocksPool }): Set<string> => {
+  let result: Set<string> = new Set();
+
+  Object.keys(blockPools).forEach((pk) => {
```

Comment:
:scream: this seems quite complex. Have you tried seeing if there is a more effective algorithm for this?